File: lib/models/appModels/Event.dart
// Event.dart
class Event {
  String? name;
  List<JoinConditions>? joinConditions;  // You will define this JoinConditions class similarly
  bool? isMainEvent;
  List<EventUserParticipant>? userParticipants;  // You will define this EventUserParticipant class similarly
  Price? price;  // You can use the Price class defined above
  List<Request>? requests;  // You will define this Request class similarly
  List<Organization>? coorganizations;  // You will define this Organization class similarly
  List<Sponsor>? sponsors;  // You will define this Sponsor class similarly
  List<Rating>? ratings;  // You will define this Rating class similarly
  List<Image>? images;  // You will define this Image class similarly
  List<Location>? location;  // You will define this Location class similarly
  EventType? type;  // You will define this EventType class similarly
  List<Payment>? payments;  // You will define this Payment class similarly
  List<Chat>? chats;  // You will define this Chat class similarly
  List<Team>? teams;  // You will define this Team class similarly
  List<Game>? games;  // You will define this Game class similarly
  List<User>? users;  // You will define this User class similarly
  List<Tryout>? tryouts;  // You will define this Tryout class similarly
  List<League>? leagues;  // You will define this League class similarly
  List<Tournament>? tournaments;  // You will define this Tournament class similarly
  List<Training>? trainings;  // You will define this Training class similarly
  List<Wager>? wager;  // You will define this Wager class similarly
  List<Rsvp>? rsvp;  // You will define this Rsvp class similarly
  bool? archived;
  bool? deleted;
  String? startTime;
  String? endTime;
  String? createdAt;
  int? capacity;
  String? mainImageKey;
  List<SocialMediaApp>? socialMediaApps;  // You will define this SocialMediaApp class similarly

  Event({
    this.name,
    this.joinConditions,
    this.isMainEvent,
    this.userParticipants,
    this.price,
    this.requests,
    this.coorganizations,
    this.sponsors,
    this.ratings,
    this.images,
    this.location,
    this.type,
    this.payments,
    this.chats,
    this.teams,
    this.games,
    this.users,
    this.tryouts,
    this.leagues,
    this.tournaments,
    this.trainings,
    this.wager,
    this.rsvp,
    this.archived,
    this.deleted,
    this.startTime,
    this.endTime,
    this.createdAt,
    this.capacity,
    this.mainImageKey,
    this.socialMediaApps,
  });
}

File: apollo/src/Event.ts
import mongoose from 'mongoose';
import {EventType} from './enums.js'; 

const Schema = mongoose.Schema;


const eventSchema = new mongoose.Schema({
    name: String,
    games: [{ type: Schema.Types.ObjectId, ref: 'Game' }],
    joinConditions: [{ type: Schema.Types.ObjectId, ref: 'JoinConditions' }],
    isMainEvent: Boolean,
    amenities: String,
    userParticipants: [{ type: Schema.Types.ObjectId, ref: 'EventUserParticipant' }],
    price: { type: Schema.Types.ObjectId, ref: 'Price' },
    requests: [{ type: Schema.Types.ObjectId, ref: 'Request' }],
    images: [{ type: Schema.Types.ObjectId, ref: 'Image' }],
    fieldLocations: [{ type: Schema.Types.ObjectId, ref: 'FieldLocation' }],
    type: {
        type: String,
        enum: ['TRYOUT', 'GAME', 'LEAGUE', 'TOURNAMENT', 'TRAINING'],
        required: true
    },
    payments: [{ type: Schema.Types.ObjectId, ref: 'Payment' }],
    chats: [{ type: Schema.Types.ObjectId, ref: 'Chat' }],
    users: [{ type: Schema.Types.ObjectId, ref: 'User' }],
    rsvp: [{ type: Schema.Types.ObjectId, ref: 'RSVP' }],
    archived: Boolean,
    deleted: Boolean,
    createdAt: String,
    startTime: String,
    endTime: String,
    capacity: Number,
    mainImageKey: String,
    coupons: [{ type: Schema.Types.ObjectId, ref: 'Coupon' }],
    eventRatings: [{ type: Schema.Types.ObjectId, ref: 'EventRating' }],    
    hostAmenities: String


    // ... other fields ...
});

const Event = mongoose.model('Event', eventSchema);

export default Event;